roles/ansible-test-splitter/files/test_split_targets.py

```diff
@@ -34,6 +59,42 @@ def test_is_slow():
     assert is_slow(["noslow", "# reason: noslow"]) is False
 
 
+@patch.object(PosixPath, "glob", MagicMock(return_value=target_dir))
+def test_get_all_targets():
+    assert get_all_targets("somewhere") == {
+        "t1": ["cloud/aws", "slow"],
+        "t2": ["t1"],
+        "t3": ["cloud/aws", "", "disabled"],
+    }
+
+
+def test_evaluate_redirection():
+    _in = {
+        "t1": ["cloud/aws", "slow"],
+        "t2": ["t1"],
+        "t3": ["cloud/aws", "", "disabled"],
+    }
+    out = {
```

Comment:
I don't think I understand what evaluate_redirection is supposed to be doing. Why aren't we including t2 in the results? Isn't that going to lead to targets being skipped that shouldn't be?

Reply:
I'm as confused as @gravesm.

Is this backwards...
If t1 is an alias of t2, then the t2 tests should be running if t1 or t2 is specified as the target...

Reply:
`t2` is an alias for `t1`. We include `t1` in the result. If we also include `t2`, it's like having two times `t1`. And we may dschedule t1 on two different jobs.

Reply:
The input is the equivalent of:
t1/aliases:
```
cloud/aws
slow
```

t2/aliases:
```
t1
```

t3/aliases:
```
cloud/aws
disabled
```

Which is a little strange...  What's more common would be

t2/aliases:
```
t4
```

And then args_targets (below) containing "t4" rather than "t1", "t2" or "t3".

Reply:
Thanks. I was reading the redirection declaration in the wrong way.

```
<tremble> goneri, I'm a little confused as to what usecase ansible/ansible-zuul-jobs#1096 is trying to cover, it feels like your test case is trying to cover a use-case we never see, but I may be misunderstanding...
<goneri> tremble, I was looking for an example and I think it helps me to understand my mistake.
<goneri> here, ec2_eni_info is an alias for ec2_eni: https://github.com/ansible-collections/amazon.aws/blob/main/tests/integration/targets/ec2_eni/aliases
<goneri> and I misunderstood how the aliases are defined. I thought there where in the opposite direction. Something like: echo ec2_eni > tests/integration/targets/ec2_eni_info/aliases
```

Reply:
I will close this PR and reopen a new one with no code change and just the valid unit-tests.

Reply:
With the following alias file: https://github.com/ansible-collections/community.aws/blob/main/tests/integration/targets/aws_config/aliases
```
$ python split_targets.py -c /home/mgraves/git/ansible_collections/community/aws/ -t aws_config_rule
{"data": {"zuul": {"child_jobs": []}, "child": {"targets_to_test": []}}}
```
I would expect this to include `aws_config` in the targets to test, right?

Reply:
Yes

> With the following alias file: https://github.com/ansible-collections/community.aws/blob/main/tests/integration/targets/aws_config/aliases
> 
> ```
> $ python split_targets.py -c /home/mgraves/git/ansible_collections/community/aws/ -t aws_config_rule
> {"data": {"zuul": {"child_jobs": []}, "child": {"targets_to_test": []}}}
> ```
> 
> I would expect this to include `aws_config` in the targets to test, right?

Yes. This is what I misunderstood.